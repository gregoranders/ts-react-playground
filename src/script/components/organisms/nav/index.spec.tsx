import React, { mount, create } from '@app/testUtils';

import { MemoryRouter } from 'react-router-dom';

import * as TestSubject from '@organisms/nav';

describe(`${TestSubject.Nav.displayName}`, () => {
  it('exports default', () => {
    expect(TestSubject.default).toBeDefined();
  });

  const TestComponent = TestSubject.default;

  it('snapshot', (): void => {
    const snapshot = create(
      <MemoryRouter>
        <TestComponent />
      </MemoryRouter>,
    );
    snapshot.toJSON();
    expect(snapshot).toMatchSnapshot();
  });

  it('mount', () => {
    const testSubject = mount(
      <MemoryRouter>
        <TestComponent />
      </MemoryRouter>,
    );
    expect(testSubject.text()).toBe('HomeMaterial UIAbout');
  });
});
