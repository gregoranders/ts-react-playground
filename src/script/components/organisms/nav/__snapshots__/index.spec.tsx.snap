// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Nav snapshot 1`] = `
<nav
  className="w3-bar w3-theme-d3"
>
  <a
    aria-current={null}
    className="w3-bar-item w3-button w3-mobile w3-hover-theme"
    href="/home"
    onClick={[Function]}
  >
    Home
  </a>
  <a
    aria-current={null}
    className="w3-bar-item w3-button w3-mobile w3-hover-theme w3-right"
    href="/about"
    onClick={[Function]}
  >
    About
  </a>
</nav>
`;
